coc/events.py

```diff
@@ -363,6 +365,115 @@ async def wrapped(cached_war: ClanWar, war: ClanWar, callback):
                 await callback(war)
         return _ValidateEvent.shortcut_register(wrapped, tags, custom_class, retry_interval, WarEvents.event_type)
 
+@_ValidateEvent
+class RaidEvents:
+    """Class that defines all valid raid events"""
+    event_type = "raid"
+
+    @classmethod
+    def new_member(cls, tags=None, custom_class=None, retry_interval=None):
+        """Event for when a new member has joined the raid weekend."""
+
+        async def wrapped(cached_raid: RaidLogEntry, raid: RaidLogEntry, callback):
+            if cached_raid.members:
+                new_members = (member for member in raid.members if member not in set(cached_raid.members))
+            else:
+                new_members = raid.members
+
+            for member in new_members:
+                await callback(member, raid)
+
+        return _ValidateEvent.shortcut_register(wrapped, tags, custom_class, retry_interval, RaidEvents.event_type)
+
+    @classmethod
+    def offensive_opponent(cls, tags=None, custom_class=None, retry_interval=None):
+        """Event for a new offensive opponent."""
+
+        async def wrapped(cached_raid: RaidLogEntry, raid: RaidLogEntry, callback):
+            if cached_raid.attack_log:
+                new_clans = (clan for clan in raid.attack_log if clan not in set(cached_raid.attack_log))
```

Comment:
Similar problem to the previous one. the clan's `__eq__` includes the attack count.